src/core/dev_api/shape_util.hpp

```diff
@@ -7,6 +7,31 @@
 #include "openvino/core/shape.hpp"
 
 namespace ov {
+
+template <class TContainer, class TAxes>
+TContainer reduce_container(const TContainer& input, const TAxes& axes) {
```

Comment:
Why do we need to add these new API to developer API?
Why cannot it be internal API?

Reply:
My intension was to provide this template to create this reduction for different type of containers or in same template implementation use it as template. But I think is not required to be public right now, I can hide it and consider to make it public if will be required.

Will update it.